GitHubReleases.ps1

```diff
@@ -0,0 +1,186 @@
+
+    .EXAMPLE
+        Get-GitHubRelease -OctoCat OctoCat
+
+    .EXAMPLE
+        Get-GitHubRelease -Uri https://github.com/PowerShell/PowerShellForGitHub
+
+    .EXAMPLE
+        Get-GitHubRelease -OrganizationName PowerShell
+
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(
+            ParameterSetName='Elements')]
+        [Parameter(
+            ParameterSetName="Elements-ReleaseId")]
+        [Parameter(
+            ParameterSetName="Elements-Latest")]
+        [Parameter(
+            ParameterSetName="Elements-Tag")]
+        [string] $OwnerName,
+
+        [Parameter(
+            ParameterSetName='Elements')]
+        [Parameter(
+            ParameterSetName="Elements-ReleaseId")]
+        [Parameter(
+            ParameterSetName="Elements-Latest")]
+        [Parameter(
+            ParameterSetName="Elements-Tag")]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-ReleaseId")]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-Latest")]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-Tag")]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Elements-ReleaseId")]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-ReleaseId")]
+        [string] $ReleaseId,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Elements-Latest")]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-Latest")]
+        [switch] $Latest,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Elements-Tag")]
+        [Parameter(
+            Mandatory,
+            ParameterSetName="Uri-Tag")]
+        [string] $Tag,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog -Invocation $MyInvocation
+
+    $elements = Resolve-RepositoryElements -BoundParameters $PSBoundParameters -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{}
+
+    $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+    $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+    $uriFragment = "repos/$OwnerName/$RepositoryName/releases"
+    $description = "Getting releases for $OwnerName/$RepositoryName"
+
+    if(-not [String]::IsNullOrEmpty($ReleaseId))
+    {
+        $telemetryProperties['ReleaseId'] = Get-PiiSafeString -PlainText $ReleaseId
```

Comment:
For these, I think it's less interesting to store the encrypted value of the input, but more so that the input was provided (simply in terms of tracking _how_ a command is being used).  So, I'd suggest changing it to this (and similar for the other two below):

```suggestion
        $telemetryProperties['ProvidedReleaseId'] = $true
```

Reply:
Thank you for pointing that out! It makes complete sense.